app/models/vm_or_template.rb

```diff
@@ -1904,6 +1904,14 @@ def self.reconfigurable?(ids)
     vms.all?(&:reconfigurable?)
   end
 
+  def self.tenant_id_clause(user_or_group)
+    template_tenant_ids = MiqTemplate.accessible_tenant_ids(user_or_group, Rbac::TENANT_ACCESS_STRATEGY["MiqTemplate"])
+    vm_tenant_ids       = Vm.accessible_tenant_ids(user_or_group, Rbac::TENANT_ACCESS_STRATEGY["Vm"])
+    return if template_tenant_ids.empty? && vm_tenant_ids.empty?
+
+    ["(template = true AND tenant_id IN (?)) OR (template = false AND tenant_id IN (?))", template_tenant_ids, vm_tenant_ids]
```

Comment:
Forget what I said...I read template_tenant_ids as "template_ids" and ran with it :(